Constraint `gson` in build classpath

wzieba opened the pull request:
Because version shipped with `violation-comments-to-github-gradle-plugin` reports a security vulnerability:

- https://github.com/wordpress-mobile/WordPress-Android/security/dependabot/32

Anyway, we should probably consider dropping support for this plugin (it's no longer maintained). I've added an item: https://github.com/orgs/Automattic/projects/841/views/6?pane=issue&itemId=74192621

### Testing

Not needed - I've added a temporary commit with dependencies change and the comment was added as expected:

<img width="927" alt="image" src="https://github.com/user-attachments/assets/de148c06-2bfd-46fb-86d3-c47f75d74852">

wzieba committed:
Constraint `gson` in build classpath

Because version shipped with `violation-comments-to-github-gradle-plugin` reports a security vulnerability

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

wzieba committed:
temp: test if reporting dependency diff GH comments still work

wzieba committed:
temp: test if reporting dependency diff GH comments still work

wzieba committed:
Revert "temp: test if reporting dependency diff GH comments still work"

This reverts commit cd6d276927f6ce2197a6d84b0d528b4b26989123.

ParaskP7 reviewed (approved):
👋 @wzieba !

I have reviewed this PR and everything seems to be working as expected, good job! 🌟 

PS: Thanks for testing it yourself with those 2 extra commits. 🥇 

-----

`EXTRAS`

1. Question (❓): This configuration applies on the module level, just because only `WordPress` is using this plugin. Just as an FYI, imaging that we would need to apply this plugin to multiple modules, in your opinion, where would we put that configuration in that case, maybe in `settings.gradle` via `resolutionStrategy`, wdyt? I am asking because maybe, in order to keep every other module as simple as possible, we should anyway put that there, at the base/root, even if only 1 module is using this plugin and such a specific configuration... 🤔

wzieba committed:
Resolve `violation-comments` plugin in main project build script

Because it's now resolved in the main build script, and not in the build script of `WordPress` module, it'll use dependencies provided by all other plugins that are resolved in the main build script as well.

So, in this particular case, `gson` in version `2.10.1` will be used for violations plugin, because `androidx.databinding:databinding-compiler-common:8.5.1` uses gson `2.10.1` and Gradle always selects the highest version.

Previously, it was not preferring `2.10.1`, because violations plugin was only resolved in `WordPress` module, and its build script was resolved in isolation from the main build script.

wzieba commented:
hi @ParaskP7 👋 your comment motivated me to try some different approaches and I found a neat solution. It's available here: 0db9460 with explanation in commit message. I'm testing it now

wzieba commented:
Works as expected!

<details><summary>Details</summary>
<p>

<img width="947" alt="image" src="https://github.com/user-attachments/assets/cc86e790-228e-4db7-99ec-6c029deda6ab">

</p>
</details>

wzieba committed:
Revert "temp: test if reporting dependency diff GH comments still work"

This reverts commit e4daf3791ae9820ef4395a6b23636a280dedde2c.

ParaskP7 commented:
Awesome, thanks for trying that and making the change! ❤️ 🚀 

Although... to be honest... this is kind of hidden now, as it is only being solved because of another plugin (`databinding`) having a newest version of `gson`... 🤷 No matter, I think we should consider this as one of the solutions nevertheless... 👍 

Still approved! ✅
